[MINOR][PYSPARK][DOCS] Fix typo in example documentation

mstill3 opened the pull request:
### What changes were proposed in this pull request?

I propose that we change the example code documentation to call the proper function .
For example, under the `foreachBatch` function, the example code was calling the `foreach()` function by mistake.

### Why are the changes needed?

I suppose it could confuse some people, and it is a typo

### Does this PR introduce any user-facing change?

No, there is no "meaningful" code being change, simply the documentation

### How was this patch tested?

I made the change on a fork and it still worked

mstill3 committed:
Fix typo in example documentation

Under the `foreachBatch` function, the example code was calling the `foreach()` function by mistake

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented:
ok to test

HyukjinKwon commented:
Can you check if there are some more typos while we're here? I am sure there are more.

SparkQA commented:
**[Test build #112873 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112873/testReport)** for PR 26299 at commit [`bd10f25`](https://github.com/apache/spark/commit/bd10f25ea1cf3716f006eb4d04eb61196be07f0f).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17800/
Test PASSed.

SparkQA commented:
**[Test build #112873 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112873/testReport)** for PR 26299 at commit [`bd10f25`](https://github.com/apache/spark/commit/bd10f25ea1cf3716f006eb4d04eb61196be07f0f).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112873/
Test FAILed.

mstill3 committed:
Fixed up some grammar issues

Fixed minor documentation issues with grammar

mstill3 committed:
Fixed typo

Fixed slight documentation typo with cube function

mstill3 committed:
Fix typo

Changed temp to temporary in documentation to conform with the rest of the documentation

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17854/
Test PASSed.

SparkQA commented:
**[Test build #112931 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112931/testReport)** for PR 26299 at commit [`f7d77e7`](https://github.com/apache/spark/commit/f7d77e758035514fb1a5c0dab7240997b7eb012f).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

mstill3 committed:
Update default values

Change all default True and False values to consistent ``True`` or ``False``

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112934/
Test FAILed.

mstill3 committed:
Fix documentation example typo

The save method in this class did not have a proper example using the save function

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17859/
Test PASSed.

SparkQA commented:
**[Test build #112936 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112936/testReport)** for PR 26299 at commit [`bc61744`](https://github.com/apache/spark/commit/bc61744d38dd91324faf6b082e8680b189dda84e).

SparkQA commented:
**[Test build #112936 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112936/testReport)** for PR 26299 at commit [`bc61744`](https://github.com/apache/spark/commit/bc61744d38dd91324faf6b082e8680b189dda84e).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112936/
Test FAILed.

srowen commented on python/pyspark/sql/session.py:
```diff
@@ -94,12 +94,12 @@ def config(self, key=None, value=None, conf=None):
             >>> from pyspark.conf import SparkConf
             >>> SparkSession.builder.config(conf=SparkConf())
-            <pyspark.sql.session...
+            <pyspark.sql.session.SparkSession.Builder at 0x7fa346504c50>
```
Hm, I don't think you want to try to match this in the doctest? that's what the ... was for, I thought

mstill3 replied:
Good call

srowen commented on python/pyspark/sql/dataframe.py:
```diff
@@ -1077,7 +1077,7 @@ def sort(self, *cols, **kwargs):
 
         :param cols: list of :class:`Column` or column names to sort by.
-        :param ascending: boolean or list of boolean (default True).
+        :param ascending: boolean or list of boolean (default ``True``).
```
What does the double-back-tick do, out of curiosity?

mstill3 replied:
It formats the True to show up as slightly different text that is for python code expressions

srowen reviewed (changes requested):
Yep several good catches in there

mstill3 committed:
Change all the data frames to DataFrame

Make the use of the word DataFrame consistent across documentation

AmplabJenkins commented:
Merged build finished. Test PASSed.

mstill3 commented on python/pyspark/sql/session.py:
```diff
@@ -624,20 +624,20 @@ def createDataFrame(self, data, schema=None, samplingRatio=None, verifySchema=Tr
         If schema inference is needed, ``samplingRatio`` is used to determined the ratio of
         rows used for schema inference. The first row will be used if ``samplingRatio`` is ``None``.
 
         :param data: an RDD of any kind of SQL data representation(e.g. row, tuple, int, boolean,
```
okay can do!

SparkQA commented:
**[Test build #112938 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112938/testReport)** for PR 26299 at commit [`c283d4f`](https://github.com/apache/spark/commit/c283d4f0d461a01869037112816531ed2d70b8e6).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112938/
Test FAILed.

mstill3 committed:
Update session.py

mstill3 committed:
Update session.py

mstill3 committed:
Add spacing in front of new sentence line

SparkQA commented:
**[Test build #112940 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112940/testReport)** for PR 26299 at commit [`85dfe22`](https://github.com/apache/spark/commit/85dfe226e1643dc8ae20116b62d92e97deedd775).

SparkQA commented:
**[Test build #112940 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112940/testReport)** for PR 26299 at commit [`85dfe22`](https://github.com/apache/spark/commit/85dfe226e1643dc8ae20116b62d92e97deedd775).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17863/
Test PASSed.

mstill3 committed:
Fixed my line length issues

Some of the lines were > 100

SparkQA commented:
**[Test build #112953 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112953/testReport)** for PR 26299 at commit [`b37f765`](https://github.com/apache/spark/commit/b37f765d01082440da3609d75d37fbdb2c70befb).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17875/
Test PASSed.

mstill3 committed:
Remove trailing whitespaces

SparkQA commented:
**[Test build #112957 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112957/testReport)** for PR 26299 at commit [`9bbd649`](https://github.com/apache/spark/commit/9bbd649eb45556cd3090dc4869b591da7c6ac715).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17879/
Test PASSed.

SparkQA commented:
**[Test build #112957 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112957/testReport)** for PR 26299 at commit [`9bbd649`](https://github.com/apache/spark/commit/9bbd649eb45556cd3090dc4869b591da7c6ac715).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112957/
Test FAILed.

mstill3 committed:
Fix note spacing issue

mstill3 committed:
Remove improper indent

SparkQA commented:
**[Test build #112958 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112958/testReport)** for PR 26299 at commit [`81d0855`](https://github.com/apache/spark/commit/81d0855673b507b6adafaf5d4058c224fea80b1d).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17880/
Test PASSed.

SparkQA commented:
**[Test build #112958 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112958/testReport)** for PR 26299 at commit [`81d0855`](https://github.com/apache/spark/commit/81d0855673b507b6adafaf5d4058c224fea80b1d).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112958/
Test FAILed.

dongjoon-hyun commented on python/pyspark/sql/readwriter.py:
```diff
@@ -733,7 +733,7 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
         :param options: all other string options
 
-        >>> df.write.mode('append').parquet(os.path.join(tempfile.mkdtemp(), 'data'))
+        >>> df.select("name", "favorite_color").write.save("namesAndFavColors.parquet")
```
This is a code change which should not be mixed into here.
Please revert this change.

srowen replied:
I think it's a fix - this is doc for `.save` and this example doesn't match

dongjoon-hyun replied:
Oh. Got it. Thanks!

dongjoon-hyun commented on python/pyspark/sql/session.py:
```diff
@@ -136,7 +136,7 @@ def appName(self, name):
         def enableHiveSupport(self):
             """Enables Hive support, including connectivity to a persistent Hive metastore, support
-            for Hive serdes, and Hive user-defined functions.
+            for Hive SerDe's, and Hive user-defined functions.
```
? This should not be a possessive form. Please revert this change.

srowen replied:
SerDes is fine though

dongjoon-hyun commented on python/pyspark/sql/session.py:
```diff
@@ -151,7 +151,7 @@ def getOrCreate(self):
 
             This method first checks whether there is a valid global default SparkSession, and if
-            yes, return that one. If no valid global default SparkSession exists, the method
+            there is, return that one. If no valid global default SparkSession exists, the method
```
If you change this from `yes` to `there is`, it breaks the remaining `If no`.
I'd like to recommend not to do this in this PR.

srowen replied:
How about just "If not," to avoid repeating the predicate.

dongjoon-hyun commented on python/pyspark/sql/streaming.py:
```diff
@@ -1062,7 +1062,7 @@ def foreachBatch(self, func):
         ...     batch_df.collect()
         ...
-        >>> writer = sdf.writeStream.foreach(func)
+        >>> writer = sdf.writeStream.foreachBatch(func)
```
Nice catch!

mstill3 replied:
Thanks!

dongjoon-hyun commented:
Thank you for your first contribution, @mstill3 .  I left a few minor comments.
BTW, please don't change the title. There is a rule for that. :)

We use `[SPARK-JIRAID][COMPONENT-NAME]` style. Since you don't make a JIRA issue, we use `[MINOR]` instead of `[SPARK-JIRAID]`.

mstill3 committed:
Made recommended changes back

mstill3 committed:
Improve wording and plural form

SparkQA commented:
**[Test build #112961 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112961/testReport)** for PR 26299 at commit [`8dbc400`](https://github.com/apache/spark/commit/8dbc40064b15ee8eface2a226f26264bdc31dd14).

SparkQA commented:
**[Test build #112961 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112961/testReport)** for PR 26299 at commit [`8dbc400`](https://github.com/apache/spark/commit/8dbc40064b15ee8eface2a226f26264bdc31dd14).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112961/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17885/
Test PASSed.

dongjoon-hyun commented:
Could you run `dev/lint-python` and fix the errors?
```
pycodestyle checks failed:
80
./python/pyspark/sql/session.py:154:97: W291 trailing whitespace
81
./python/pyspark/sql/session.py:155:98: W291 trailing whitespace
82
##[error]Process completed with exit code 1.
```

mstill3 committed:
Remove trailing whitespace

SparkQA commented:
**[Test build #112964 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112964/testReport)** for PR 26299 at commit [`3dea837`](https://github.com/apache/spark/commit/3dea8372de3236a654c7699030751ad75b7b31bd).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17887/
Test PASSed.

SparkQA commented:
**[Test build #112964 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112964/testReport)** for PR 26299 at commit [`3dea837`](https://github.com/apache/spark/commit/3dea8372de3236a654c7699030751ad75b7b31bd).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112964/
Test FAILed.

mstill3 committed:
Add space before parentheses

SparkQA commented:
**[Test build #112967 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112967/testReport)** for PR 26299 at commit [`21cc7ea`](https://github.com/apache/spark/commit/21cc7ea5b81f3e3e1f881ae6b78fcdd2e0766a17).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17890/
Test PASSed.

dongjoon-hyun commented on python/pyspark/sql/dataframe.py:
```diff
@@ -1975,7 +1980,7 @@ def withColumn(self, colName, col):
 
         The column expression must be an expression over this DataFrame; attempting to add
-        a column from some other dataframe will raise an error.
+        a column from some other DataFrame will raise an error.
```
Maybe, did you want the following?
```
a column from some other :class:`DataFrame` will raise an error.
```

mstill3 replied:
Yeah good catch

dongjoon-hyun commented on python/pyspark/sql/session.py:
```diff
@@ -151,9 +151,9 @@ def getOrCreate(self):
-            yes, return that one. If no valid global default SparkSession exists, the method
-            creates a new SparkSession and assigns the newly created SparkSession as the global
-            default.
+            there is, return that one. Otherwise if no valid global default SparkSession exists,
```
`Otherwise if`? Shall we keep the original sentences for this part?

SparkQA commented:
**[Test build #112967 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112967/testReport)** for PR 26299 at commit [`21cc7ea`](https://github.com/apache/spark/commit/21cc7ea5b81f3e3e1f881ae6b78fcdd2e0766a17).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112967/
Test FAILed.

mstill3 committed:
Add missing parameter to trigger method

Added more verbosity on how it is working

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17891/
Test PASSed.

SparkQA commented:
**[Test build #112969 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112969/testReport)** for PR 26299 at commit [`43b6f2b`](https://github.com/apache/spark/commit/43b6f2ba18037e0eec58930e7f5f772bf7070d52).

SparkQA commented:
**[Test build #112969 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112969/testReport)** for PR 26299 at commit [`43b6f2b`](https://github.com/apache/spark/commit/43b6f2ba18037e0eec58930e7f5f772bf7070d52).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112969/
Test FAILed.

mstill3 committed:
Contistent use of DataFrames in documentation

mstill3 committed:
Change the syntax back to how it was

SparkQA commented:
**[Test build #112970 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112970/testReport)** for PR 26299 at commit [`4eb3537`](https://github.com/apache/spark/commit/4eb35376683463c83f4e9e94b41a05ec86583cc4).

SparkQA commented:
**[Test build #112970 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112970/testReport)** for PR 26299 at commit [`4eb3537`](https://github.com/apache/spark/commit/4eb35376683463c83f4e9e94b41a05ec86583cc4).
 * This patch **fails Python style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112970/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17893/
Test PASSed.

mstill3 committed:
Set documentation length to correct width

<= 100 characters per line

SparkQA commented:
**[Test build #112971 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112971/testReport)** for PR 26299 at commit [`cef3f12`](https://github.com/apache/spark/commit/cef3f1203c979f757ab6daabaa6e84487ee3c0c2).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17894/
Test PASSed.

SparkQA commented:
**[Test build #112971 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112971/testReport)** for PR 26299 at commit [`cef3f12`](https://github.com/apache/spark/commit/cef3f1203c979f757ab6daabaa6e84487ee3c0c2).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112971/
Test FAILed.

dongjoon-hyun commented on python/pyspark/sql/readwriter.py:
```diff
@@ -733,7 +733,7 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
         :param options: all other string options
 
-        >>> df.write.mode('append').parquet(os.path.join(tempfile.mkdtemp(), 'data'))
+        >>> df.select("name", "favorite_color").write.save("namesAndFavColors.parquet")
```
@mstill3 . As you see, this causes UT failure because `df` has only `[day, month, name, year]`.
```
Failed example:
    df.select("name", "favorite_color").write.save("namesAndFavColors.parquet")
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib64/pypy-2.5.1/lib-python/2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest pyspark.sql.readwriter.DataFrameWriter.save[0]>", line 1, in <module>
        df.select("name", "favorite_color").write.save("namesAndFavColors.parquet")
      File "/home/jenkins/workspace/SparkPullRequestBuilder@2/python/pyspark/sql/dataframe.py", line 1325, in select
        jdf = self._jdf.select(self._jcols(*cols))
      File "/home/jenkins/workspace/SparkPullRequestBuilder@2/python/lib/py4j-0.10.8.1-src.zip/py4j/java_gateway.py", line 1286, in __call__
        answer, self.gateway_client, self.target_id, self.name)
      File "/home/jenkins/workspace/SparkPullRequestBuilder@2/python/pyspark/sql/utils.py", line 102, in deco
        raise converted
    AnalysisException: cannot resolve '`favorite_color`' given input columns: [day, month, name, year];;
```

dongjoon-hyun reviewed (changes requested):
The UT failure seems to be the last blocker for this PR. Could you fix that in this PR?

mstill3 committed:
Fix example in documentation 

update save() example code to match the DF columns

SparkQA commented:
**[Test build #112973 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112973/testReport)** for PR 26299 at commit [`0207d87`](https://github.com/apache/spark/commit/0207d87d6c612bccfc3608729375430dc0bba7e8).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17896/
Test PASSed.

SparkQA commented:
**[Test build #112973 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112973/testReport)** for PR 26299 at commit [`0207d87`](https://github.com/apache/spark/commit/0207d87d6c612bccfc3608729375430dc0bba7e8).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112973/
Test FAILed.

dongjoon-hyun commented on python/pyspark/sql/readwriter.py:
```diff
@@ -733,7 +733,7 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
         :param options: all other string options
 
-        >>> df.write.mode('append').parquet(os.path.join(tempfile.mkdtemp(), 'data'))
+        >>> df.select("day", "month", "name", "year").write.save("mydata.parquet")
```
I guess we need `mode("overwrite")`.
```
File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/readwriter.py", line 735, in pyspark.sql.readwriter.DataFrameWriter.save
Failed example:
    df.select("day", "month", "name", "year").write.save("mydata.parquet")
Exception raised:
    Traceback (most recent call last):
      File "/home/anaconda/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest pyspark.sql.readwriter.DataFrameWriter.save[0]>", line 1, in <module>
        df.select("day", "month", "name", "year").write.save("mydata.parquet")
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/readwriter.py", line 746, in save
        self._jwrite.save(path)
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/lib/py4j-0.10.8.1-src.zip/py4j/java_gateway.py", line 1286, in __call__
        answer, self.gateway_client, self.target_id, self.name)
      File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/utils.py", line 102, in deco
        raise converted
    AnalysisException: path file:/home/jenkins/workspace/SparkPullRequestBuilder/mydata.parquet already exists.;
```

dongjoon-hyun replied:
BTW, let's remove `.select("day", "month", "name", "year")`. We don't need to `select` because this is `save` API doc.

mstill3 committed:
Fix documentation example 

Changed save mode to overwrite in save() documentation example

SparkQA commented:
**[Test build #113045 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113045/testReport)** for PR 26299 at commit [`0d5fc4d`](https://github.com/apache/spark/commit/0d5fc4d296bb740a45a73ea68e1a045bbd5d4c8d).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17966/
Test PASSed.

SparkQA commented:
**[Test build #113045 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113045/testReport)** for PR 26299 at commit [`0d5fc4d`](https://github.com/apache/spark/commit/0d5fc4d296bb740a45a73ea68e1a045bbd5d4c8d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/113045/
Test PASSed.

HyukjinKwon commented on python/pyspark/sql/readwriter.py:
```diff
@@ -733,7 +733,7 @@ def save(self, path=None, format=None, mode=None, partitionBy=None, **options):
         :param options: all other string options
 
-        >>> df.write.mode('append').parquet(os.path.join(tempfile.mkdtemp(), 'data'))
+        >>> df.write.mode("overwrite").save("mydata.parquet")
```
I think we should revert this change. Tmp directory will be created under target directory as of https://github.com/apache/spark/commit/a634d66ce767bd5e1d8553d1a2c32e2b1a80f642 ; however, now seems it will create under the current working directory.

srowen replied:
The change is really to use `.save`, as that what this is supposed to be an example of. If it makes sense to make a temp file to save to, yes I agree, leave that part the same.

HyukjinKwon replied:
Oh, yes. We should use `.save`. Thanks for clarification.

HyukjinKwon reviewed (approved):
LGTM otherwise.

dongjoon-hyun commented:
@mstill3 . Please update the PR to use `os.path.join(tempfile.mkdtemp(), 'data'` according to @HyukjinKwon 's advice.

dongjoon-hyun reviewed (approved):
LGTM for the others.

mstill3 committed:
Change to saving in temp directory

Update save() documentation example to create the file in the temp directory instead of in the root

SparkQA commented:
**[Test build #113099 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113099/testReport)** for PR 26299 at commit [`4e909e2`](https://github.com/apache/spark/commit/4e909e2241feedc6506d5396b533d1d3faf66d8d).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/18015/
Test PASSed.

SparkQA commented:
**[Test build #113099 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113099/testReport)** for PR 26299 at commit [`4e909e2`](https://github.com/apache/spark/commit/4e909e2241feedc6506d5396b533d1d3faf66d8d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/113099/
Test PASSed.

dongjoon-hyun reviewed (approved):
Thank you again for your first contribution, @mstill3 .
Thank you, @srowen, @HyukjinKwon , and @vegerot
Merged to master.

dongjoon-hyun commented:
@mstill3 . Since this is useful for users, could you make a PR for `branch-2.4` too?
